src/coreclr/inc/winwrap.h

```diff
@@ -254,112 +254,14 @@ WszCreateProcess(
     LPPROCESS_INFORMATION lpProcessInformation
     );
 
-#if defined(HOST_X86) && defined(_MSC_VER)
-
-//
-// Windows SDK does not use intrinsics on x86. Redefine the interlocked operations to use intrinsics.
-//
-
-#include "intrin.h"
-
-#define InterlockedIncrement            _InterlockedIncrement
-#define InterlockedDecrement            _InterlockedDecrement
-#define InterlockedExchange             _InterlockedExchange
-#define InterlockedCompareExchange      _InterlockedCompareExchange
-#define InterlockedExchangeAdd          _InterlockedExchangeAdd
-#define InterlockedCompareExchange64    _InterlockedCompareExchange64
-#define InterlockedAnd                  _InterlockedAnd
-#define InterlockedOr                   _InterlockedOr
+#ifdef HOST_WINDOWS
 
 //
-// There is no _InterlockedCompareExchangePointer intrinsic in VC++ for x86.
-// winbase.h #defines InterlockedCompareExchangePointer as __InlineInterlockedCompareExchangePointer,
-// which calls the Win32 InterlockedCompareExchange, not the intrinsic _InterlockedCompareExchange.
-// We want the intrinsic, so we #undef the Windows version of this API, and define our own.
+// Workaround for https://github.com/microsoft/WindowsAppSDK/issues/4074
+// Windows SDK is missing InterlockedCompareExchange8 definition.
 //
-#ifdef InterlockedCompareExchangePointer
-#undef InterlockedCompareExchangePointer
-#endif
-
-FORCEINLINE
-PVOID
-InterlockedCompareExchangePointer (
-    __inout  PVOID volatile *Destination,
-    _In_opt_ PVOID ExChange,
-    _In_opt_ PVOID Comperand
-    )
-{
-    return((PVOID)(LONG_PTR)_InterlockedCompareExchange((LONG volatile *)Destination, (LONG)(LONG_PTR)ExChange, (LONG)(LONG_PTR)Comperand));
-}
-
-#endif // HOST_X86 && _MSC_VER
-
-#if defined(HOST_X86) & !defined(InterlockedIncrement64)
-
-// Interlockedxxx64 that do not have intrinsics are only supported on Windows Server 2003
```

Comment:
I am wondering whether the removal of this Windows Server 2003 workaround can be introducing the libatomic dependendency.

Reply:
I thought this header is windows only?

Reply:
CoreCLR Win32 PAL is a partial Windows emulator. A lot of things that look Windows-specific in CoreCLR are actually cross-plat.

Submitted #97358 to check

Reply:
Yes, it is the problem. You can add the `#if defined(HOST_X86) & !defined(InterlockedIncrement64)` block back to fix the Linux x86 build breaks and revert the libatomic linking.

Reply:
https://github.com/dotnet/runtime/pull/97358 is merged.

Reply:
I'll update the PR and fix the tests later today.